### ISSUE TITLE
Pytorch Lightning integration

### ISSUE THREAD

**Issue by EgorKraevTransferwise** [sdv-dev/SDV#413]
It would be great to have the models use PyTorch Lightning under the hood, it would allow to use standard Lightning tooling for tracking the progress of the fit, automatically suggesting a good learning rate, enabling multi-GPU training, etc.

If I were to submit refactors of eg TVAE to preserve current interfaces, but use Lightning under the hood, would these be accepted?

**Comment by npatki** [comment]
Hi @EgorKraevTransferwise, the team is actively working on improving the interfaces, models and abstractions to make the library easier to use. It might be best if we wait for the changes to land before embarking on refactors.

I'll keep this issue open as a feature request. If you're still interested in making a contribution, we can reach back out when we're at a more stable state.

**Comment by EgorKraevTransferwise** [comment]
Sounds good! We're not using SDV at the moment, but might well come back to that in a quarter or two, and if so we'll want to train the models on datasets of 100Ms of rows, so looks like the timing might line up
